File: python/linear_search.py
"""
This algorithm has a linear runtime because in the worse case scenario the algorithm has to compare the target to every item in the list.
"""

def linear_search(list, target):
  """
  Returns the index position of the target if found, else returns None
  """
  
  for i in range(0, len(list)):
    if list[i] == target:
      return i
  return None
 
def verify(index):
  if index is not None:
    print("Target found at index: ", index)
  else:
    print("Target not found in list")
    
numbers = [1,2,3,4,5,6,7,8,9,10]

result = linear_search(numbers, 12)
verify(result)

result = linear_search(numbers,6)
verify(result)

File: javascript/linear_search.js
function linear_search(list, target) {
	// go through each item in the list and compare it to the target.
	for(var i = 0; i < list.length; i++) {
		// when the index in the list matches the target return the index.
		if(list[i] == target) {
			return i;
		}
	}
	// In case the target is not found then return -1
	return -1;
}

function verify(index) {
	if(index !== -1) {
		console.log('Target found at index: ' + index);
	} else {
		console.log('Target not found!');
	}
}


var list = [];

for(var  i = 0; i <= 100000000; i++) {
	list.push(i);
}

verify(linear_search(list, 10));
verify(linear_search(list, 200));
verify(linear_search(list, 10000));
